ports/zephyr/CMakeLists.txt

```diff
@@ -143,3 +145,62 @@ target_sources(app PRIVATE
 )
 
 target_link_libraries(app PRIVATE ${MICROPY_TARGET})
+
+if (CONFIG_MICROPY_DYNAMIC_PINCTRL)
+    if (CONFIG_MICROPY_DYNAMIC_PINCTRL_GENERATE_PINMUX AND NOT CONFIG_MICROPY_DYNAMIC_PINCTRL_CUSTOM_GENERATOR)
+
+        find_file(MICROPY_DYNAMIC_PINCTRL_HEADER_FOUND
+        ${CONFIG_MICROPY_DYNAMIC_PINCTRL_HEADER}
+        ${DTS_ROOT_SYSTEM_INCLUDE_DIRS})
+
+        set(MICROPY_GENERATED_DYNAMIC_PINCTRL "${CMAKE_CURRENT_BINARY_DIR}/generated_dynamic_pinctrl.h")
+
+        string(REPLACE ";" ";-I;" DTS_ROOT_SYSTEM_INCLUDE_DIRS_STRING "-I${DTS_ROOT_SYSTEM_INCLUDE_DIRS}")
+
+        # Executed at configuration time
+        execute_process(
+        COMMAND ${CMAKE_C_COMPILER}
+            ${DTS_ROOT_SYSTEM_INCLUDE_DIRS_STRING}
+            -dM -E ${MICROPY_DYNAMIC_PINCTRL_HEADER_FOUND}
+        OUTPUT_VARIABLE MICROPY_PROCESSED_DYNAMIC_PINCTRL_HEADER_DATA
+        COMMAND_ERROR_IS_FATAL ANY
+        )
+
+        message("Using regex: ${CONFIG_MICROPY_DYNAMIC_PINCTRL_REGEX} for Pinctrl module generation")
+        string(REGEX MATCHALL ${CONFIG_MICROPY_DYNAMIC_PINCTRL_REGEX} MICROPY_PROCESSED_DYNAMIC_PINCTRL_HEADER_DATA_REGEXED ${MICROPY_PROCESSED_DYNAMIC_PINCTRL_HEADER_DATA})
```

Comment:
In case you were after suggestions for shorter names:

`MICROPY_PROCESSED_DYNAMIC_PINCTRL_HEADER_DATA`
`MICROPY_DYNAMIC_PINCTRL_DATA`

`MICROPY_PROCESSED_DYNAMIC_PINCTRL_HEADER_DATA_REGEXED`
`MICROPY_DYNAMIC_PINCTRL_REGEXED`

This is just resources to pick from, not a recommendation, long descriptive names are useful too.

Reply:
I think it's necessary there.